lib/scholar/linear/logistic_regression.ex

```diff
@@ -0,0 +1,195 @@
+defmodule Scholar.Linear.LogisticRegression do
+  @moduledoc """
+  Logistic regression in both binary and multinomial variants
+  """
+  import Nx.Defn
+  alias __MODULE__
+
+  @derive {Nx.Container, containers: [:coefficients, :bias], keep: [:mode]}
+  defstruct [:coefficients, :bias, :mode]
+
+  @doc """
+  Fits a logistic regression model for sample inputs `x` and
+  sample targets `y`. Depending on number of classes the function chooses
+  either binary or multinomial logistic regression.
+  """
+  defn fit(x, y, opts \\ []) do
+    fit_verify(x, y, opts)
+    if opts[:num_classes] < 3, do: fit_binary(x, y, opts), else: fit_multinomial(x, y, opts)
+  end
+
+  # Function checks validity of the provided data
+
+  deftransformp fit_verify(x, y, opts) do
+    if !is_integer(opts[:num_classes]) or opts[:num_classes] < 1 do
+      raise ArgumentError, "The number of classes must be a positive integer"
+    end
+
+    if Nx.rank(x.shape) != 2 do
+      raise ArgumentError, "Training vector must be two-dimensional (n_samples, n_features)"
+    end
+
+    if (Nx.rank(y.shape) != 1 and !opts[:one_hot]) or (Nx.rank(y.shape) != 2 and opts[:one_hot]) do
+      raise ArgumentError,
+            "Target vector must be one-dimensional (n_samples) or two-dimensional if :one_hot set to true"
+    end
```

Comment:
Maybe instead of requiring the user pass `:one_hot` we could determine it based on whether `y` is rank-1 or rank-2?

Also, is there a good reason for supporting one-hot here? The reason why the user would have the data in one-hot is if they encoded as such in the first place, or is there another reason? Is this something `scikit` does?

Reply:
Well, I tried to test my algorithms on data with a text-categorical label. In Explorer, at this point, I cannot find something like OrdinalEncoder from SciKit. But Explorer got dummies, so I use them. Frankly, when I think about this now, I don't think we need to offer this option, and I can get rid of it.

Reply:
But this ordinalEncoder is just dummies + argmax